### PR TITLE
Add torch.serialization.safe_globals context manager

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Add context manager mentioned in https://github.com/pytorch/pytorch/pull/127808#pullrequestreview-2096298486

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #127939

### CODE REVIEW

File: test/test_serialization.py [hunk 1]
```diff
@@ -4194,8 +4194,6 @@ def test_safe_globals_for_weights_only(self):
         '''
         Tests import semantic for tensor subclass and the {add/get/clear}_safe_globals APIs
         '''
-        # Needed to prevent UnboundLocalError: local variable 'TwoTensor' referenced before assignment
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:4197]: Sorry this should have been removed in the below PR

**Review by albanD — APPROVED** [review]
Sounds good!

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**